BrightID/src/components/HomeScreen.js

```diff
@@ -10,284 +9,333 @@ import {
   View,
   TextInput,
 } from 'react-native';
+import { SvgXml } from 'react-native-svg';
 import ActionSheet from 'react-native-actionsheet';
-import { connect } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import Material from 'react-native-vector-icons/MaterialCommunityIcons';
 import Ionicons from 'react-native-vector-icons/Ionicons';
 import { setPhoto, setName } from '@/actions';
 import { getNotifications } from '@/actions/notifications';
-import { delStorage } from '@/utils/dev';
 import { chooseImage, takePhoto } from '@/utils/images';
 import { saveImage, retrieveImage } from '@/utils/filesystem';
-import { DEVICE_TYPE } from '@/utils/constants';
+import { DEVICE_LARGE, DEVICE_IOS } from '@/utils/constants';
 import fetchUserInfo from '@/actions/fetchUserInfo';
-import VerificationSticker from './Verifications/VerificationSticker';
+import verificationSticker from '@/static/verification-sticker.svg';
+
+if (__DEV__) {
+  const { delStorage } = require('@/utils/dev');
+}
+
 /**
  * Home screen of BrightID
  * ==========================
  */
 
-type State = {
-  profilePhoto: string,
-  isEditing: boolean,
-  name: string,
-};
-
-let chatSheetRef = '';
+let chatSheetRef = '',
+  photoSheetRef = '';
 let discordUrl = 'https://discord.gg/nTtuB2M';
+let JoinCommunity = DEVICE_IOS ? TextInput : Text;
 
-export class HomeScreen extends React.Component<Props, State> {
-  photoSheetRef: string;
+export const HomeScreen = (props) => {
+  const { navigation } = props;
+  const dispatch = useDispatch();
+  const id = useSelector((state) => state.user.id);
+  const name = useSelector((state) => state.user.name);
+  const photoFilename = useSelector((state) => state.user.photo.filename);
+  const groups = useSelector((state) => state.groups.groups);
+  const apps = useSelector((state) => state.apps.apps);
+  const verifications = useSelector((state) => state.user.verifications);
+  const connections = useSelector((state) => state.connections.connections);
+  const [profilePhoto, setProfilePhoto] = useState('');
+  const [isEditing, setIsEditing] = useState(false);
+  const [displayName, setDisplayName] = useState(name);
 
-  // eslint-disable-next-line react/state-in-constructor
-  state = {
-    profilePhoto: ' ',
-    isEditing: false,
-    name: '',
-  };
+  const verified = useMemo(() => verifications.includes('BrightID'), [
+    verifications,
+  ]);
 
-  componentDidMount() {
-    const { navigation, dispatch, photo, name } = this.props;
+  useEffect(() => {
```

Comment:
You could refactor this to make use of the react-navigation hooks. (https://reactnavigation.org/docs/use-focus-effect)

Reply:
thanks, I'm using that effect on a different page, and should probably use it here as well